include/ideep/operators/inner_product.hpp

```diff
@@ -526,7 +537,11 @@ struct inner_product_forward
       } else {
         dst.feed_from(expected_dst);
       }
-    } else { // reorder_src
+    } else {
+      if (!reorder_src)  {
```

Comment:
This check is not necessary?

Reply:
I think it is necessary, which make sure the user gives a dst(will use it's format ) when reorder_src is false.

Reply:
> LGTM except for some small nits. Does the "matmul" primitive have sijmilar issue?

matmul doesn't have such an issue, it always uses input's desc to create primitive.